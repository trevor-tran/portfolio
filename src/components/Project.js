import ArrowOutwardIcon from '@mui/icons-material/ArrowOutward';

import "./Project.scss";

export default function Project({ project }) {

  const { title, url, descriptions, skills, date, image } = project;

  let concatenatedDescription = "";
  descriptions.forEach((bullet) => {
    concatenatedDescription = concatenatedDescription + bullet + " ";
  });

  return (
    <div
      className={"row " + (project.url ? "project-container-link" : "project-container-no-link")}
      onClick={() => project.url ? window.open(project.url, '_blank') : null}>
      {image ?
        <div className="col-12 col-xl-4 order-last order-xl-first">
          <img className="img-thumbnail rounded" src={require(`../assets/${image}`)} alt="app preview image" />
        </div>
        :
        <div className="col-12 col-xl-4 date">
          {date.start.toUpperCase()} <span>&#8212;</span> {date.end.toUpperCase()}
        </div>
      }
      <div className="col">
        <div className="d-flex">
          <p className="h5">{title}</p>
          {url && <span><ArrowOutwardIcon className="arrow" fontSize="small" /></span>}
        </div>
        <p>{concatenatedDescription}</p>
        <div className="d-flex flex-wrap">
          {skills.map((skill, i) =>
            <div className="skill" key={i}>{skill}</div>)}
        </div>
      </div>
    </div>
  )
}