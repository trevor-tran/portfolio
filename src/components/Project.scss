@mixin project-container($hasLink: false) {
  padding-top: 10px;
  padding-bottom: 10px;
  margin-bottom: 40px;

  &:hover {
    border-radius: 10px;
    background-color: rgba(217, 217, 217, 0.1);
    @if $hasLink {
      cursor: pointer;
    } @else {
      cursor: default;
    }
    .arrow {
      transform: translate(5px, -5px);
    }
  }

}
.project-container-no-link {
  @include project-container($hasLink: false);
}

.project-container-link {
  @include project-container($hasLink: true);
}

.skill {
  color: #39275b;
  margin-top: 7px;
  margin-right: 7px;
  background-color: rgba(215,200,150, 1);
  padding: 2px 5px 2px 5px;
  border-radius: 20px;
}
