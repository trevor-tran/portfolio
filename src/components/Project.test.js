import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Project from './Project';

describe("Project component", () => {

  it("verifies against its snapshot", () => {
    const mockProject = {
      title: "title",
      company: "firmABC",
      date: {
        start: "startDate",
        end: "endDate"
      },
      descriptions: ["description1", "description2"],
      skills: ["skill1", "skill2"]
    };
    const tree = renderer
      .create(<Project project={mockProject} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
