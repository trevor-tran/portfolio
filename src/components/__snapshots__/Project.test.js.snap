// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component verifies against its snapshot 1`] = `
<div
  className="row project-container-no-link"
  onClick={[Function]}
>
  <div
    className="col-12 col-xl-4 date"
  >
    STARTDATE
     
    <span>
      â€”
    </span>
     
    ENDDATE
  </div>
  <div
    className="col"
  >
    <div
      className="d-flex"
    >
      <p
        className="h5"
      >
        title
      </p>
    </div>
    <p>
      description1 description2 
    </p>
    <div
      className="d-flex flex-wrap"
    >
      <div
        className="skill"
      >
        skill1
      </div>
      <div
        className="skill"
      >
        skill2
      </div>
    </div>
  </div>
</div>
`;
